tools/preprocess.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# SPDX-FileCopyrightText: © 2024 ZeldaRET
+# SPDX-License-Identifier: CC0-1.0
+
+# Usage: preprocess [flags] -- [compile command minus input file...] [single input file]
+# Flags: -v OOT_VERSION (required)
+# Preprocess a C file to:
+# * Re-encode from UTF-8 to EUC-JP
+#   (the repo uses UTF-8 for text encoding, but the strings in the ROM are encoded in EUC-JP)
+# * Replace `#pragma increment_block_number` (see preprocess_pragma)
+
+set -e
+set -o pipefail
+
+if [ "${VERBOSE-}" ]
+then
+    set -x
+fi
+
+for i in `seq ${#@}`
+do
+    if [[ "${!i}" = '--' ]]
+    then
+        # flags before --
+        flags=("${@:1:$(($i - 1))}")
+        # compile command, betwen -- and the input source file
+        compilecmd="${@:$(($i + 1)):$((${#@} - $i - 1))}"
+        # The last argument, the input source file to be compiled
+        srcfile="${@: -1}"
+        break
+    fi
+done
+
+if [ "${VERBOSE-}" ]
+then
+    echo flags="${flags[@]}"
+    echo compilecmd="$compilecmd"
+    echo srcfile="$srcfile"
+fi
+
+while getopts "v:" opt "${flags[@]}"
+do
+    case $opt in
+        v)
+            OOT_VERSION=$OPTARG
+            ;;
+        ?)
+            echo "Error: Bad flags"
+            exit 1
+            ;;
+    esac
+done
+
+if [[ "${!OPTIND}" != '--' ]]
+then
+    echo "Error: Positional arguments in flags not allowed"
+    exit 1
+fi
+
+if [ -z "${OOT_VERSION-}" ]
+then
+    echo Missing -v
+    exit
+fi
+
+# Create a temporary file, and remove it on script exit
+tempfile=`mktemp`_oot.c
```

Comment:
ido_block_numbers.py and fix_bss.py are currently broken with this change, since relied on this tempfile having the same basename as the original source file, because ido writes the symbol table as `filename.T` and I couldn't find a way to change that. That's why preprocess.py created a tempdir instead